src/main/scala/performanceanalysis/administrator/Administrator.scala

```diff
@@ -48,31 +48,25 @@ class Administrator(logReceiverActor: ActorRef) extends Server {
 
   protected def routes: Route = componentsRoute
 
-  private def handleGetComponents(resultFuture: Future[Any]): Future[HttpResponse] = {
+  private def handleGet(resultFuture: Future[Any]): Future[HttpResponse] = {
     resultFuture.flatMap {
       case RegisteredComponents(componentIds) =>
         val entityFuture = Marshal(RegisteredComponents(componentIds)).to[ResponseEntity]
-        entityFuture.map {
-          case registeredComponentsEntity =>
-            HttpResponse(
-              status = StatusCodes.OK,
-              entity = registeredComponentsEntity
-            )
-        }
+        toFutureResponse(entityFuture)
+      case Details(componentId) =>
+        val entityFuture = Marshal(Details(componentId)).to[ResponseEntity]
+        toFutureResponse(entityFuture)
     }
   }
 
-  private def handleGetDetails(resultFuture: Future[Any]): Future[HttpResponse] = {
-    resultFuture.flatMap {
-      case Details(componentId) =>
-        val entityFuture = Marshal(Details(componentId)).to[ResponseEntity]
-        entityFuture.map {
-          case registeredComponentsEntity =>
-            HttpResponse(
-              status = StatusCodes.OK,
-              entity = registeredComponentsEntity
-            )
-        }
+  private def toFutureResponse(entityFuture: Future[ResponseEntity]) = {
+    entityFuture.map {
+      case registeredComponentsEntity =>
+        HttpResponse(
+          status = StatusCodes.OK,
```

Comment:
But what if you want to respond with another statuscode, depending on the resultFuture?